packages/angular/build/src/utils/index-file/index-html-generator.ts

```diff
@@ -86,6 +92,11 @@ export class IndexHtmlGenerator {
       this.csrPlugins.push(addNgcmAttributePlugin());
       this.ssrPlugins.push(addEventDispatchContractPlugin(), addNoncePlugin());
     }
+
+    // Auto-CSP (as the last step)
+    if (options.autoCsp) {
+      this.csrPlugins.push(autoCspPlugin(options.autoCsp.unsafeEval));
```

Comment:
@alan-agius4, do we also want to push this to `this.ssrPlugins`? I think that uses a different copy of the `index.html` page which might need the same transformation, but I'm not 100% on the architecture here.

Reply:
Yeah, the SSR does use a different copy. During SSR, there are additional script tags added and removed. 

However, it seems that this will not work for SSR, as this happens because render and during SSR rendering there are extra scripts that are added/removed.

Reply:
So, to clarify, we do not have concerns around leaving the plugin as-is and only applying to `csrPlugins`, correct?

Reply:
Yeah that's fine. I'd love to find a way to make this SSR compatible in the future, but that can be a future feature.

Should we throw an error somewhere if someone configures `autoCsp` *and* enables SSR? We can also do that in a follow up PR, doesn't need to be this one.

Reply:
Seems reasonable to me.